### ISSUE TITLE
Arrow functions: lexical scope "this" not working

### ISSUE THREAD

**Issue by Avien** [babel/babel#1887]
It seems the "this" scope refers to the global window and not to the relevant object

```javascript
export class ToolBarService{

    constructor(){
        this.items = {
            'exportTo': null
        }
    }

    get export(){
        return {
            'onClick' : (handler)=>{
                this.items.exportTo = handler;
            }
        }
    }
}
```
this.items is undefined in the 'onClick' function

Important to say that the transpiled code in the bundle.js does seem to look correct:
( I am using webpack babel-loader with babel-core 5.6.15)
```javascript
_createClass(ToolBarService,[{
    key:'export',
    get:function get(){
        var _this = this;
        
        return {
            'onClick' : function onClick(handler){
                _this.items.exportTo = handler;
            }
        }
    }
}]);
```

But it doesn't work
Help please
10x

**Comment by sebmck** [comment]
As the README says, the Babel issue tracker is not for support:

> For questions and support please visit the <a href="https://babel-slack.herokuapp.com">slack channel</a> or <a href="http://stackoverflow.com/questions/tagged/babeljs">StackOverflow</a>. The Babel issue tracker is <strong>exclusively</strong> for bug reports and feature requests.

Please provide a complete reproducible example in the [REPL](http://babeljs.io/repl) if you think you've found a bug.

**Comment by Avien** [comment]
ok let me rephrase, in this line:

 this.items.exportTo = handler;

I expect the "this" to refer to the outer this which suppose to be the class "ToolBarService" and not the window object

**Comment by sebmck** [comment]
That's still not enough information. I don't know how you're interfacing with that method. Please provide a complete reproducible example.

**Comment by Avien** [comment]
Why do you need more information? when I call that 'onClick' function from another service the "this" refers to the local scope 'onClick' function and not the this defined in the constructor

I just refactored all my code from traceur in it works fine there

**Comment by sebmck** [comment]
Because it matters how you're interfacing with that method. Just because it worked with Traceur doesn't mean it should. If I can't reproduce this then I can't debug why it's happening and what the expected behaviour is.
